pybaseball/team_batting_against.py

```diff
@@ -0,0 +1,49 @@
+from typing import Optional
+
+import pandas as pd
+
+from . import cache
+from .datahelpers import postprocessing
+from .datasources.bref import BRefSession
+from .utils import most_recent_season
+
+session = BRefSession()
+
+# pylint: disable=line-too-long
+_URL = "https://www.baseball-reference.com/leagues/majors/{year}-batting-pitching.shtml#teams_batting_pitching"
+
+
+def get_team_batting_against(year: int) -> pd.DataFrame:
+    url = _URL.format(year=year)
+    res = session.get(url, timeout=None)
+    if res.ok:
+        team_batting_against = pd.read_html(res.content)
+        return team_batting_against
+    else:
+        raise ValueError(res.reason)
+
+
+@cache.df_cache()
+def team_batting_against(season: Optional[int] = None) -> pd.DataFrame:
+    if season is None:
+        season = most_recent_season()
+    team_batting_against = get_team_batting_against(season)
+    team_batting_against = pd.concat(team_batting_against)
+    team_batting_against = postprocess(team_batting_against)
+    return team_batting_against
+
+
+def postprocess(team_batting_against: pd.DataFrame) -> pd.DataFrame:
+    # drop usually empty PA for which data is not known column
+    team_batting_against = team_batting_against.drop('PAu', 1)
+
+    # skip league average and duplicative header
+    team_batting_against = team_batting_against.iloc[:30]
```

Comment:
This fails in the case of not having 30 teams. E.g. 

```
In [5]: team_batting_against(1950)
Out[5]:
                        Tm  RA/G     G     PA     AB      R      H  ...     TB   GDP  HBP    SH  SF  IBB   ROE
0           Boston Red Sox  5.22   154   6086   5224    804   1413  ...   2109   140   23    90   0   18    57
1         Brooklyn Dodgers  4.67   155   5996   5336    724   1397  ...   2187   150   21    46   0   37    65
2            Boston Braves  4.72   156   6024   5365    736   1411  ...   2074   108   30    75   0   29    82
3             Chicago Cubs  5.01   154   6061   5364    772   1452  ...   2157   113   27    76   0   35   108
4        Chicago White Sox  4.80   156   6058   5215    749   1370  ...   1957   123   21    88   0   39    63
5          Cincinnati Reds  4.80   153   5956   5267    734   1363  ...   2133   109   41    64   0   57    93
6        Cleveland Indians  4.22   155   5948   5201    654   1289  ...   1893   131   18    82   0   39    76
7           Detroit Tigers  4.54   157   6077   5400    713   1444  ...   2116   131   31    93   0   32    61
8          New York Giants  4.18   154   5802   5160    643   1268  ...   1981   136   31    75   0   39    91
9         New York Yankees  4.46   155   6002   5187    691   1322  ...   1940   136   35    72   0   10    62
10  Philadelphia Athletics  5.93   154   6168   5324    913   1528  ...   2276   144   30    85   0   17    75
11   Philadelphia Phillies  3.97   157   5953   5346    624   1324  ...   2002   119   13    66   0   39    90
12      Pittsburgh Pirates  5.56   154   6076   5349    857   1472  ...   2214   120   32    73   0   55    67
13        St. Louis Browns  5.95   154   6285   5528    916   1629  ...   2262    91   32    74   0   23    74
14     St. Louis Cardinals  4.38   153   5872   5229    670   1398  ...   2060   118   25    81   0   52    69
15     Washington Senators  5.25   155   6110   5327    813   1479  ...   2134   140   26   109   0   30   102
16          League Average  4.85   155   6030   5301    751   1410  ...   2093   126   27    78   0   34    77
17                     NaN  4.85  2476  96474  84822  12013  22559  ...  33495  2009  436  1249   0  551  1235

[18 rows x 26 columns]
```

You should be able to do `team_batting_against.iloc[:-2]` to get the same effect.

I would also kind of prefer league average row to stay on there (e.g. `.iloc[:-1]` instead). Any specific reason to cut it?

Reply:
> Any specific reason to cut [the league average row]?
No, just too focused on my intended use case. I'll see that its included.